### PR TITLE
Add support for inheritance

### LINKED ISSUE #24: Native inheritance/base classes and multi inheritance
Hey! Great library.

Is it possible currently to define a native C++ class's base/parent class? Furthermore, is it possible for that definition to include the parents base class and so on. I noticed an in-progress/todo commented function `base`, I tried to use it and it seemed to work to some extent but there are issues. The prototype properties seem to be there form the parent, but using any of those methods or properties causes an exception "ParentClass object expected"

In essence I would like to do:
```
        module.class_<Point>("Point")
                .constructor<int, int>()
                .fun<&Point::x>("x")
                .fun<&Point::y>("y")
                .fun<&Point::norm>("norm");

        module.class_<Point3D>("Point3D")
                .base<Point>()
                .constructor<int, int, int>()
                .fun<&Point3D::z>("z")
```

where
```
class Point3D : public Point { }
```

then in JS:
```
const point = new Point3D(1, 2, 3);
console.log('point xyz', point.x, point.y, point.z);
```

### PR DESCRIPTION
**Pull request by jprendes**
This is an attempt to add inheritance to `quickjspp`.
Rather than skipping the type verification, as discussed in #24, this change adds a mechanism to cast the pointer stored in the opaque to the correct type.
It relies on an unordered_map, but the cost of look-up is only paid when actually using inheritance.
This change should close #24

### CODE REVIEW

File: quickjspp.hpp [hunk 1]
```diff
@@ -783,12 +851,26 @@ struct js_traits<std::shared_ptr<T>>
...
-    {
-        auto ptr = static_cast<std::shared_ptr<T> *>(JS_GetOpaque2(ctx, v, QJSClassId));
+    static const std::shared_ptr<T> unwrap(JSContext * ctx, JSValueConst v)
+    {
+        std::shared_ptr<T> ptr = nullptr;
+        auto obj_class_id = JS_GetClassID(v);
+
+        if (obj_class_id == QJSClassId) {
+            // The JS object is of class T
+            ptr = *static_cast<std::shared_ptr<T> *>(JS_GetOpaque2(ctx, v, obj_class_id));
```
Comment (ftk) [inline quickjspp.hpp:861]:
```suggestion
            void * opaque = JS_GetOpaque2(ctx, v, obj_class_id);
            assert(opaque && "No opaque pointer in object");
            ptr = *static_cast<std::shared_ptr<T> *>(opaque);
```

File: quickjspp.hpp [hunk 2]
```diff
@@ -498,7 +500,7 @@ struct unwrap_arg_impl {
     {
         if (size_t(argc) <= I) {
             JS_ThrowTypeError(ctx, "Expected type %lu arguments but only %d were provided",
-                              (unsigned long)NArgs, argc);
+                              NArgs, argc);
```
Comment (ftk) [inline quickjspp.hpp:503]: https://github.com/ftk/quickjspp/runs/4259581835?check_suite_focus=true#step:5:89

**Review by ftk — APPROVED** [review]

File: quickjspp.hpp (hunk 1 above)
Comment (jprendes) [reply quickjspp.hpp:861]: That's a good catch, thanks!

File: quickjspp.hpp (hunk 2 above)
Comment (jprendes) [reply quickjspp.hpp]: Missed that. Should be fixed now.

**Pull request merged**